core/dialog.ts

```diff
@@ -108,14 +119,38 @@ export function prompt(
  */
 export function setPrompt(
   promptFunction: (
-    p1: string,
-    p2: string,
-    p3: (p1: string | null) => void,
-  ) => void,
+    message: string,
+    defaultValue: string,
+    callback: (result: string | null) => void,
+  ) => void = defaultPrompt,
 ) {
   promptImplementation = promptFunction;
 }
 
-export const TEST_ONLY = {
-  confirmInternal,
-};
+/**
+ * Displays a temporary notification atop the workspace. Blockly provides a
+ * default toast implementation, but developers may provide their own via
+ * setToast. For simple appearance customization, CSS should be sufficient.
+ *
+ * @param workspace The workspace to display the toast notification atop.
+ * @param options Configuration options for the notification, including its
+ *     message and duration.
+ */
+export function toast(workspace: WorkspaceSvg, options: ToastOptions) {
+  toastImplementation(workspace, options);
+}
+
+/**
+ * Sets the function to be run when Blockly.dialog.toast() is called.
+ *
+ * @param toastFunction The function to be run.
```

Comment:
can you add "or undefined to restore default implementation" to all of the tsdoc?

Reply:
Done.